mmedit/models/extractors/feedback_hour_glass.py

```diff
@@ -0,0 +1,90 @@
+import torch.nn as nn
+
+
+class ResBlock(nn.Module):
+    """ResBlock for HourGlass.
+
+    It has a style of:
+
+    ::
+
+        ---Conv-ReLU-Conv-Conv-+-
+         |_________Conv________|
+
+        or
+
+        ---Conv-ReLU-Conv-Conv-+-
+         |_____________________|
+
+    Args:
+        in_channels (int): Number of channels in the input features.
+        out_channels (int): Number of channels in the output features.
+    """
+
+    def __init__(self, in_channels, out_channels):
+        super().__init__()
+        self.conv_block = nn.Sequential(
+            nn.Conv2d(in_channels, out_channels // 2, 1),
+            nn.ReLU(inplace=True),
+            nn.Conv2d(
+                out_channels // 2, out_channels // 2, 3, stride=1, padding=1),
+            nn.Conv2d(out_channels // 2, out_channels, 1))
+        if in_channels == out_channels:
+            self.skip_layer = None
+        else:
+            self.skip_layer = nn.Conv2d(in_channels, out_channels, 1)
+
+    def forward(self, x):
+        """Forward function.
+
+        Args:
+            x (Tensor): Input tensor with shape (n, c, h, w).
+
+        Returns:
+            Tensor: Forward results.
+        """
+
+        residual = self.conv_block(x)
+        if self.skip_layer:
+            x = self.skip_layer(x)
+        return x + residual
+
+
+class HourGlass(nn.Module):
+    """Hour Glass model for face landmark.
+
+    It is a recursive model.
+
+    Args:
+        depth (int): Depth of HourGlass, the number of recursions.
+        mid_channels (int): Number of channels in the intermediate features.
+    """
+
+    def __init__(self, depth, mid_channels):
+        super().__init__()
+        self.up1 = ResBlock(mid_channels, mid_channels)
```

Comment:
Is this an upsampling?

Reply:
No

Reply:
Give it a better name :)

Reply:
In https://github.com/open-mmlab/mmpose/blob/master/mmpose/models/backbones/hourglass.py, it is called `up1`

Reply:
Digged a little about the history. mmpose's version comes from mmdet https://github.com/open-mmlab/mmdetection/pull/2840.

It is a bad naming. However since it has been existed in multiple mm-repos, let's keep it for now.